scene/gui/option_button.cpp

```diff
@@ -118,12 +118,13 @@ void OptionButton::_notification(int p_what) {
 			}
 
 			Size2 size = get_size();
+			Ref<StyleBox> style = _get_current_stylebox();
 
 			Point2 ofs;
 			if (is_layout_rtl()) {
-				ofs = Point2(theme_cache.arrow_margin, int(Math::abs((size.height - theme_cache.arrow_icon->get_height()) / 2)));
+				ofs = Point2(theme_cache.arrow_margin + style->get_margin(SIDE_LEFT), int(Math::abs((size.height - theme_cache.arrow_icon->get_height() - style->get_minimum_size().height) / 2 + style->get_margin(SIDE_TOP))));
```

Comment:
Consider the currently used stylebox when calculating the position of the arrow icon